metals/src/main/scala/scala/meta/internal/metals/ClientCommands.scala

```diff
@@ -170,6 +171,51 @@ object ClientCommands {
     """.stripMargin
   )
 
+  val UpdateTestExplorer = new ListParametrizedCommand[BuildTargetUpdate](
+    "metals-update-test-explorer",
+    "Update Test Explorer",
+    "Notifies the client that the test explorer model has to be updated",
+    """
+      |Sends to the client an Array of `BuildTargetUpdate`s.
+      |
+      |```ts
+      |export interface BuildTargetUpdate {
+      |  targetName: TargetName;
+      |  targetUri: TargetUri;
+      |  events: TestExplorerEvent[];
+      |}
+      |export type TestExplorerEvent =
+      | RemoveTestSuite
+      | AddTestSuite
+      | AddTestCases;
+      |
+      |interface BaseTestExplorerEvent {
+      |  fullyQualifiedClassName: FullyQualifiedClassName;
+      |  className: ClassName;
+      |}
+      |export interface RemoveTestSuite extends BaseTestExplorerEvent {
```

Comment:
We don't need extend data here I think? We can just explicitely write what each event provides.

Reply:
It's copypasted from typescript's file, I didn't too much attention to such details

Reply:
Would be cool if we could specify it all explictely, but not super high priority.

Reply:
Specified explicitly in `ClientCommands`